packages/browser/lib/core/crypto.ts

```diff
@@ -227,6 +295,27 @@ export class CoreCrypto {
       ["encrypt"]
     );
 
+    let version: string;
+    let dataToEncrypt: Uint8Array;
+    if (role != null) {
+      version = this.#config.versions.LCY;
+      const metadataBytes = CoreCrypto.#buildMetadata(
+        Math.floor(Date.now() / 1000),
+        role
+      );
+      const metadataOffset = new Uint8Array(2); // Metadata size as a 2 bytes little-endian unsigned integer
+      metadataOffset[0] = metadataBytes.length & 0xff;
+      metadataOffset[1] = (metadataBytes.length >> 8) & 0xff;
+      dataToEncrypt = concatUint8Arrays([
+        metadataOffset,
+        metadataBytes,
+        utf8StringToUint8Array(str),
+      ]);
+    } else {
+      version = this.#config.versions.NOC;
+      dataToEncrypt = utf8StringToUint8Array(str);
+    }
```

Comment:
Do we only assign metadata if a role is present in the other SDK's? Given the `buildMetadata` function doesn't require a role I think it might make sense to always assign metadata to have the encrypted at timestamp.

Reply:
We should be doing this everywhere but some SDKs dont. Not sure why/where that diverged but I'll add it here since we get more benefit out of it.